pynestml/utils/mechs_info_enricher.py

```diff
@@ -0,0 +1,178 @@
+                            update_expr_str)
+                        # pretend that update expressions are in "equations" block,
+                        # which should always be present, as differential equations
+                        # must have been defined to get here
+                        update_expr_ast.update_scope(
+                            neuron.get_equations_blocks()[0].get_scope())
+                        update_expr_ast.accept(ASTSymbolTableVisitor())
+
+                        solution_transformed["states"][variable_name] = {
+                            "ASTVariable": variable,
+                            "init_expression": expression,
+                            "update_expression": update_expr_ast,
+                        }
+                    for variable_name, rhs_str in ode_info["ode_toolbox_output"][ode_solution_index][
+                        "propagators"].items(
+                    ):
+                        prop_variable = neuron.get_equations_blocks()[0].get_scope().resolve_to_symbol(variable_name,
+                                                                                                  SymbolKind.VARIABLE)
+                        if prop_variable is None:
+                            ASTUtils.add_declarations_to_internals(
+                                neuron, ode_info["ode_toolbox_output"][ode_solution_index]["propagators"])
+                            prop_variable = neuron.get_equations_blocks()[0].get_scope().resolve_to_symbol(
+                                variable_name,
+                                SymbolKind.VARIABLE)
+
+                        expression = ModelParser.parse_expression(rhs_str)
+                        # pretend that update expressions are in "equations" block,
+                        # which should always be present, as synapses have been
+                        # defined to get here
+                        expression.update_scope(
+                            neuron.get_equations_blocks()[0].get_scope())
+                        expression.accept(ASTSymbolTableVisitor())
+
+                        solution_transformed["propagators"][variable_name] = {
+                            "ASTVariable": prop_variable, "init_expression": expression, }
+                        expression_variable_collector = ASTEnricherInfoCollectorVisitor()
+                        expression.accept(expression_variable_collector)
+
+                        neuron_internal_declaration_collector = ASTEnricherInfoCollectorVisitor()
+                        neuron.accept(neuron_internal_declaration_collector)
+
+                        # print("TRV: " + PredefinedFunctions.TIME_RESOLUTION)
+                        for variable in expression_variable_collector.all_variables:
+                            for internal_declaration in neuron_internal_declaration_collector.internal_declarations:
+                                # print(internal_declaration.get_variables()[0].get_name())
```

Comment:
remove; you can also use logging debug print for text that could be meaningful to the user